### ISSUE TITLE
CanCan/matchers doesn't work when multiple permissions are checked at the same time

### ISSUE THREAD

**Issue by douglascamata** [CanCanCommunity/cancancan#362]
I'm using the `be_able_to` matcher available at `cancan/matchers` to check if my Ability is able to do many actions on a model and it fails. Curious thing: if I try them all manually, every single one of them work. 

This is the Ability: 

```ruby
class Ability
  include CanCan::Ability

  def initialize(user)
    can [:write, :rate, :read], Job, user_id: user.id
  end
end

```

And this is my test:

```ruby
require "rails_helper"

RSpec.describe Ability do
  subject { Ability.new(user) }
  let(:user) { create(:user) }

  it "should be able to write, rate and read his own job" do
    permissions = [:write, :rate, :read]
    permissions.each do |p|
      expect(subject).to be_able_to(
        p,
        Job.new(driver_id: user.id),
        driver_id: user.id
      )
    end

    expect(subject).to be_able_to(
      permissions,
      Job.new(driver_id: user.id),
      driver_id: user.id
    )
  end
end
```

So, what happens when I run this spec is:

- first assertion group (in the `permissions.each` loop) is ok, all assertions pass
- second assertion (using the permissions array) fails

They are testing the same thing and should all pass. 

## Versions

Rails: 4.2.7
CanCanCan: 1.13.1

**Comment by gingray** [comment]
@douglascamata if you take a look on matcher implementation you will see that it just call can?
```ruby
  match do |ability|
    ability.can?(*args)
  end
```

that means you pass array and it interpret as action name thats why your code fails. Workaround that you use i do the same way. I think about to make pull request to add another matcher which will deal with arrays of actions.

**Comment by douglascamata** [comment]
@gingray I don't think it should be a different matcher. Because if with `can` (to define abilities) I can use an array of abilities, I should be able to ask with `can?` and have a response that follow the same though process as the definition. 

If with `can [:write, :rate, :read], Job, user_id: user.id` I define that an User can write, rate and read all jobs he owns, when I use `can? [:write, :rate, :read], Job, user_id: user.id` it should check if the user can write, rate and read all jobs he owns. Which is is case would be true.

**Comment by gingray** [comment]
@douglascamata from one side you right but from other in terms of REST you shouldn't make action with two or more responsibilities. There also pull request from 2015 to add multiple can? but it still in pull request https://github.com/CanCanCommunity/cancancan/pull/262

**Comment by gingray** [comment]
I've add pull request to solve this issue https://github.com/CanCanCommunity/cancancan/pull/364

**Comment by douglascamata** [comment]
@gingray I don't think about it as something that I would actually use in the code of my app, but something that I would use in the tests for sure.

**Comment by gingray** [comment]
@douglascamata i've got your point hope pull request will be approved and we can use same method to work with array of abilities from tests.

**Comment by Rajan4436** [comment]
any solution ?? I m still facing the same issue

**Comment by coorasse** [comment]
this should have been patched by #364. please reopen if the problem persists
